src/phpDebug.ts

```diff
@@ -264,8 +273,30 @@ class PhpDebugSession extends vscode.DebugSession {
             });
             server.listen(args.port || 9000, (error: NodeJS.ErrnoException) => error ? reject(error) : resolve());
         });
+
+        /** sets up a client to register IDE to DBgp proxy */
+        const createProxyClient = () => new Promise((resolve, reject) => {
+            const client = this._proxyClient = net.createConnection({port: args.proxyPort | 9001, host: args.proxyHost}, () => {
+                this.sendEvent(new vscode.OutputEvent('DBgp proxy: connected to ' + args.proxyHost + ':' + args.proxyPort + '\n'));
+                client.write('proxyinit -p ' + args.proxyPort + ' -k ' + args.ideKey + ' -m 1');
+            });
+
+            client.on('data', (data:string) => {
+                client.end();
```

Comment:
Ok for the error handling.
I don't understand what do you mean with "all the command-queue logic should be moved from XDebugConnection" I worked only on PHPDebug.ts

From XDebug page I did not understand when/where proxy stop should be executed. Do you?